solr/core/src/java/org/apache/solr/servlet/HttpSolrCall.java

```diff
@@ -603,6 +608,44 @@ public Action call() throws IOException {
 
   }
 
+  /** Get the span for this request.  Not null. */
+  protected Span getSpan() {
+    // Span was put into the request by SolrDispatchFilter
+    return (Span) Objects.requireNonNull(req.getAttribute(Span.class.getName()));
+  }
+
+  // called after init().
+  protected void populateTracingSpan(Span span) {
+    // Set db.instance
+    String coreOrColName = HttpSolrCall.this.origCorename;
+    if (coreOrColName == null && getCore() != null) {
+      coreOrColName = getCore().getName();
+    }
+    if (coreOrColName != null) {
+      span.setTag(Tags.DB_INSTANCE, coreOrColName);
+    }
+
+    // Set operation name.
+    String path = getPath();
+    if (coreOrColName != null) {
+      // prefix path by core or collection name
+      if (getCore() != null && getCore().getName().equals(coreOrColName)) {
+        path = "/{core}" + path;
+      } else {
+        path = "/{collection}" + path;
+      }
+    }
+    String verb =
+        getQueryParams()
+            .get(CoreAdminParams.ACTION, req.getMethod())
+            .toLowerCase(Locale.ROOT);
+    span.setOperationName(verb + ":" + path);
```

Comment:
is it possible that verb setting is null? is it tolerable?

Reply:
Impossible.  See the "get" that takes 2 args.  The 2nd param there is the default, which is the "method" (HTTP) of the request, which can't be null.